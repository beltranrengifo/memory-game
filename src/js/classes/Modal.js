export default class Modal {
  constructor () {
    this.overlay = document.createElement('div')
    this.modalBody = document.createElement('div')
    this.loader = document.createElement('div')
    this.createModal()
  }

  createModal () {
    const body = document.querySelector('body')

    this.overlay.classList.add('modal__overlay')
    this.modalBody.classList.add('modal__body')

    this.overlay.appendChild(this.modalBody)
    body.appendChild(this.overlay)
  }

  showModal () {
    this.overlay.classList.add('modal__overlay--visible')
  }

  hideModal () {
    this.overlay.classList.remove('modal__overlay--visible')
  }

  showLoader () {
    this.loader = document.createElement('div')
    this.loader.classList.add('modal__loader')
    this.loader.innerHTML = '<i class="fal fa-spinner-third fa-spin"></i>'
    this.modalBody.appendChild(this.loader)
  }

  hideLoader () {
    this.modalBody.removeChild(this.loader)
  }

  setContent (content) {
    this.modalBody.innerHTML = content
  }

  deleteContent () {
    this.modalBody.innerHTML = ''
  }

  destroyModal () {
    this.hideModal()
    setTimeout(() => this.overlay.remove(), 800)
  }
}
