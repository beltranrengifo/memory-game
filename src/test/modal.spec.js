import Modal from '@/js/classes/Modal'

describe('Modal class', () => {
  let modal

  beforeEach(() => {
    modal = new Modal()
    document.body.innerHTML = `
      <div class="header__info" />
      <div id="main-board" />
    `
    modal.createModal()
  })

  test('Modal function', () => {
    expect(typeof Modal).toBe('function')
  })

  test('modal to be defined', () => {
    expect(modal).toBeDefined()
  })

  test('modal to match snapshot', () => {
    expect(modal).toMatchSnapshot()
  })

  test('spyOn createModal on new Modal()', () => {
    const spy = jest.spyOn(modal, 'createModal')
    modal.createModal()
    expect(spy).toHaveBeenCalled()
  })

  test('check overlay creation', () => {
    const a = document.body.querySelector('.modal__overlay')
    expect(a).toBeTruthy()
  })

  test('check modal body creation', () => {
    const a = document.body.querySelector('.modal__body')
    expect(a).toBeTruthy()
  })

  test('check showModal fn', () => {
    modal.showModal()
    const a = document.body.querySelector('.modal__overlay')
    const hasClass = a.classList.contains('modal__overlay--visible')
    expect(hasClass).toBe(true)
  })

  test('check hideModal fn', () => {
    modal.showModal()
    const a = document.body.querySelector('.modal__overlay')
    modal.hideModal()
    const hasClass = a.classList.contains('modal__overlay--visible')
    expect(hasClass).toBe(false)
  })

  test('check showLoader fn', () => {
    modal.showLoader()
    const modalBody = document.body.querySelector('.modal__body')
    const loader = modalBody.querySelector('.modal__loader')
    expect(loader).toBeTruthy()
  })

  test('showLoader should match snapshot', () => {
    modal.showLoader()
    const modalBody = document.body.querySelector('.modal__body')
    expect(modalBody).toMatchSnapshot()
  })

  test('check hideLoader fn', () => {
    modal.showLoader()
    const modalBody = document.body.querySelector('.modal__body')
    modal.hideLoader()
    const loader = modalBody.querySelector('.modal__loader')
    expect(loader).toBeFalsy()
  })

  test('check setContent fn', () => {
    const html = '<div class="demo-class">Hi there</div>'
    modal.setContent(html)
    expect(modal.modalBody.innerHTML).toBe(html)
  })

  test('check deleteContent fn', () => {
    const html = '<div class="demo-class">Hi there</div>'
    modal.setContent(html)
    modal.deleteContent()
    expect(modal.modalBody.innerHTML).toBe('')
  })
})
