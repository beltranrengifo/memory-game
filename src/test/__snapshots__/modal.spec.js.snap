// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal class modal to match snapshot 1`] = `
Modal {
  "loader": <div />,
  "modalBody": <div
    class="modal__body"
  />,
  "overlay": <div
    class="modal__overlay"
  >
    <div
      class="modal__body"
    />
  </div>,
}
`;

exports[`Modal class showLoader should match snapshot 1`] = `
<div
  class="modal__body"
>
  <div
    class="modal__loader"
  >
    <i
      class="fal fa-spinner-third fa-spin"
    />
  </div>
</div>
`;
